### PR TITLE
Change README to AsciiDoc

### PR DESCRIPTION
**Pull request by magicDGS**
Proposal to change the README to AsciiDoc, as it supports out-of-the-box "Table of Contents", apart of a really nice markup syntax (more consistent than Markdown).

Also, I included some changes in the content.

It also includes changes from #24:
- Badges
- Build status (including AppVeyor)

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+image:http://img.shields.io/badge/language-java-brightgreen.svg[Language, link=https://www.java.com]
+image:https://img.shields.io/badge/license-BSD%203--Clause-blue.svg[License (3-Clause BSD), link=https://opensource.org/licenses/BSD-3-Clause]
+
+Staging ground for the development of the next major version of HTSJDK
+
+toc::[]
+
+== Status
+
+[options="header", cols=2, caption=""]
```
Comment (lbergelson) [inline README.adoc]: I would probably put these as a single horizontal line or as part of the badge line.  This table looks nice but takes up most of the vertical space.

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,56 @@
+= htsjdk-next-beta
+:toc: macro
+
+image:http://img.shields.io/badge/language-java-brightgreen.svg[Language, link=https://www.java.com]
```
Comment (lbergelson) [inline README.adoc]: If we go with asciidoc we might want a footnote at the bottom with a link to an asciidoc quick reference guide.

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+== Getting started
+
+=== How to build
+
+* Linux/MacOS
++
+....
+./gradlew assemble
+....
+* Windows
```
Comment (lbergelson) [inline README.adoc]: Can asciidoc generate support those interactive panes where you can have multiple tabs for different language examples?  Something like the [panes here](https://cloud.google.com/storage/docs/access-public-data)

**Comment by tfenne** [comment]
Sorry for the delay weighing in on this.  I have to say, I'm sorry, but I'm against switching away from MarkDown.  My brain is already annoyed at having to recall both MarkDown and RST syntax, and adding yet another "not-HTML" language just to get an automated TOC seems like a bad tradeoff.  It also looks more complicated (to my eye) than MarkDown, but that might be lack of familiarity.

File: README.adoc (hunk 1 above)
Comment (magicDGS) [reply README.adoc]: Done

File: README.adoc (hunk 3 above)
Comment (magicDGS) [reply README.adoc]: I did put them in a table form similar to the build.

**Comment by magicDGS** [comment]
@tfenne - I understand you, as I was also against AsciiDoc now that I am using it but I found that not only the built-in TOC is a nice feature. I propose it for several reasons:

1. Have a consistent markup language. For writing technical documentation and other stuff that I am planning to add (see #42 as an example), I think that Markdown is really limited: using AsciiDoc and some extensions (like the diagams) we can have even embeded UML diagrams to document how the architecture of the project looks like. Having the README in one format and the rest in other format is kind of inconsistent (similar to different formatting around the code).
1. The syntax is fixed but extensible, and there are not dialects like with Markdown. This means that in every computer, it will render (better or worse depending on the command that was used). That is not the case if you are editting Markdown without specifying that it is GitHub-flavored, for example.
1. Other features apart of the TOC might be useful in the future. By now, with this simple README, it does not look like an amazing thing to do. For example, the multi-line code for the build/test tables should be done with html in markdown (see below for more about this).
1. Using AsciiDoc avoids to use a lot of hardcoded HTML as Markdown sometimes force to do. For example, to do a subscript/superscript in Markdown you are forced to use HTML (e.g., `x<sub>2</sub>` to render to x<sub>2</sub>), but in AsciiDoc it is as simple as `x~2`. For x<sup>2</sup>, where `x<sup>2</sup>` is less readable for me than `x^2`

If you are still not convinced by this, I can give up the README. But I think that we should definetely write in AsciiDoc the technical documentation and user manual on how to use the library (and also in a way that the documentation is compiled and every test snippet is run - I have a very cool idea about this). And I would really like to have technical documentation to define a proper architecture to be able to fullfill all the requirements from other projects (e.g., see #34 and #42 for a concrete proposal designed before the implementation).

**Comment by tfenne** [comment]
Closing; in person conversation with @lbergelson - we agree to stay with MarkDown.

**Pull request closed without merging**